pygmt/io.py

```diff
@@ -0,0 +1,36 @@
+"""
+PyGMT input/output (I/O) utilities.
+"""
+import xarray as xr
+
+
+def process_output_grid(grid_name, tmpfile_name):
+    """
+    Processes the output from the GMT API to return an xarray.DataArray if
+    ``grid_name`` matches ``tmpfile_name`` and return None if it does not.
```

Comment:
Based on [`xarray.load_dataarray`](https://xarray.pydata.org/en/v0.19.0/generated/xarray.load_dataarray.html) at https://github.com/pydata/xarray/blob/a21dead8621480b608827355b50e85af6f88f185/xarray/backends/api.py#L248-L255, but modified to mention the `GMTDataArrayAccessor`.
```suggestion
    Open, load into memory, and close a DataArray from a file or file-like
    object containing a single data variable.

    This is a thin wrapper around :py:meth:`xarray.open_dataarray`. It differs
    from `open_dataarray` in that it loads the DataArray into memory, gets GMT
    specific metadata about the grid via :meth:`GMTDataArrayAccessor`, closes
    the file, and returns the DataArray. In contrast, `open_dataarray` keeps
    the file handle open and lazy loads its contents. All parameters are passed
    directly to `open_dataarray`. See that documentation for further details.
```